test/Test/Database/LSMTree/Internal/RawBytes.hs

```diff
@@ -0,0 +1,86 @@
+{- HLINT ignore "Avoid restricted alias" -}
+{- HLINT ignore "Use /=" -}
+
+module Test.Database.LSMTree.Internal.RawBytes (tests) where
+
+import           Database.LSMTree.Extras.Generators ()
+import           Database.LSMTree.Internal.RawBytes (RawBytes)
+import qualified Database.LSMTree.Internal.RawBytes as RawBytes (size)
+import           Test.QuickCheck (Property, collect, mapSize, withDiscardRatio,
+                     (.||.), (===), (==>))
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.QuickCheck (testProperty)
+
+-- * Tests
+
+tests :: TestTree
+tests = testGroup "Test.Database.LSMTree.Internal.RawBytes" $
+        [
+            testGroup "Eq laws" $
+            [
+                testProperty "Reflexivity"  propEqReflexivity,
+                testProperty "Symmetry"     propEqSymmetry,
+                testProperty "Transitivity" propEqTransitivity,
+                testProperty "Negation"     propEqNegation
+            ],
+            testGroup "Ord laws" $
+            [
+                testProperty "Comparability" propOrdComparability,
+                testProperty "Transitivity"  propOrdTransitivity,
+                testProperty "Reflexivity"   propOrdReflexivity,
+                testProperty "Antisymmetry"  propOrdAntisymmetry
+            ]
+        ]
+
+-- * Utilities
+
+withFirstBlockSizeInfo :: RawBytes -> Property -> Property
+withFirstBlockSizeInfo firstBlock
+    = collect ("Size of first block is " ++ show (RawBytes.size firstBlock))
+
+-- * Properties to test
+
+-- ** 'Eq' laws
```

Comment:
With the current (and previous) method of generating `RawBytes`, it's unlikely to generate two `RawBytes` that are equal. It means we mostly test `block1 /= block2` cases. If we want, we could try to change that

Reply:
I’ve added output of statistics about block equality. It turns out that typically about 1.5 % of any two generated blocks are equal. Since this is quite a small share, I’ve increased the number of test cases for the above-mentioned properties to 10,000. All these changes are in be22c3f202fe0bca61ea94dd0d6949fb99ced9aa.